src/ansys/dpf/composites/layup_info/_add_layup_info_to_mesh.py

```diff
@@ -71,21 +76,28 @@ def add_layup_info_to_mesh(
         dictionary key in the :attr:`.ContinuousFiberCompositesFiles.composite`
         attribute. This parameter is only required for assemblies.
         See the note about assemblies in the description for the :class:`.CompositeModel` class.
+    rst_stream_provider:
+        Pass RST stream to load section data directly from the RST file. Support with DPF 2024R2
+        (8.0) and later.
+
     Returns
     -------
     :
         Lay-up provider operator.
     """
-    composite_data_sources = _get_composite_data_sources_for_layup_provider(
-        data_sources, composite_definition_label
-    )
-
     # Set up the lay-up provider.
     # Reads the composite definition file and enriches the mesh
     # with the composite lay-up information.
     layup_provider = Operator("composite::layup_provider_operator")
     layup_provider.inputs.mesh(mesh)
-    layup_provider.inputs.data_sources(composite_data_sources)
+
+    composite_data_sources = _get_composite_data_sources_for_layup_provider(
```

Comment:
the lay-up file is now optional and so passed to the lay-up provider only if it is available.